pyduofern/duofern.py

```diff
@@ -107,14 +109,14 @@ def del_device(self, code, name=None):
         if code in self.modules['by_code']:
             del self.modules['by_code'][code]
 
-    def update_state(self, code, key, value, trigger=None):
-        self.modules['by_code'][code][key] = value
+    def update_state(self, hash, key, value, trigger=None):
+        hash[key] = value
         if self.changes_callback:
             self.changes_callback()
 
-    def delete_state(self, code, key):
-        if key in self.modules['by_code'][code]:
-            del self.modules['by_code'][code][key]
+    def delete_state(self, hash, key):
+        if key in hash:
+            del hash[key]
```

Comment:
Instead of modifying the passed dict here, it may make sense to just fix the ``code`` or ``device_id`` in ``Duofern.parse()``. Therein one could introduce a ``device_id`` then every time hash is set to a new value one would have to modify ``device_id`` accordingly. Or (after reading your thoughts on device in FHEM vs device for Rademacher) one might make ``update_state`` subdevice-aware by coming up with a a subdevice id.